src/Illuminate/Database/Migrations/Migrator.php

```diff
@@ -441,16 +435,23 @@ protected function getQueries($migration, $method)
     }
 
     /**
-     * Resolve a migration instance from a file.
+     * Resolve a migration instance from migration path.
      *
-     * @param  string  $file
+     * @param  string  $path
      * @return object
      */
-    public function resolve($file)
+    protected function resolve(string $path)
```

Comment:
We can't update this to being a path. Apps will break if they expect to pass a file instead. All these changes to this method need to be reverted and a different solution needs to be implemented.

Reply:
I changed the code to have the new implementation in the `resolvePath` method. This method `resolve` was adjusted so that it is public again, and keeps the old behavior, even though it is not used in the Migrator anymore. Still, I changed the name of the parameter, because what we give to this method is actually a migration name (like "2020_01_01_000000_create_users_table"), not a file name.

I also restored the function `requireFiles`. It is also not used anymore, but in theory it is a breaking change.